state-reconstruct-fetcher/src/l1_fetcher.rs

```diff
@@ -232,13 +222,14 @@ impl L1Fetcher {
         let tx_handle = self.spawn_tx_handler(
             hash_rx,
             calldata_tx,
-            token.clone(),
+            self.cancellation_token.clone(),
```

Comment:
This is odd; if the cancellation token is included in `L1Fetcher`, it doesn't have to be passed separately - no?